test/k8s/datapath_configuration.go

```diff
@@ -810,25 +809,25 @@ func checkMonitorOutput(monitorOutput []byte, egressPktCount, ingressPktCount in
 	By("Looking for TCP notifications using the ephemeral port %q", portBytes)
 	port, err := strconv.Atoi(string(portBytes))
 	if err != nil {
-		GinkgoPrint("ephemeral port %q could not be converted to integer: %s", string(portBytes), err)
-		return false
+		return fmt.Errorf("ephemeral port %q could not be converted to integer: %s",
+			string(portBytes), err)
 	}
 
 	expEgress := fmt.Sprintf("SrcPort=%d", port)
 	expEgressRegex := regexp.MustCompile(expEgress)
 	egressMatches := expEgressRegex.FindAllIndex(monitorOutput, -1)
 	if len(egressMatches) != egressPktCount {
-		GinkgoPrint("Could not locate final FIN notification in monitor log: egressTCPMatches %+v", egressTCPMatches)
-		return false
+		return fmt.Errorf("monitor logs contained unexpected number (%d) of egress notifications matching %q",
+			len(egressMatches), expEgress)
```

Comment:
Yes, but notice the `%+v` is gone :-)

Reply:
Also, won't they be printed only after the last Eventually retry?

Reply:
> It's just not obvious to me how this formatted string gets to stdout — just asking to make sure it doesn't go to /dev/null :)

`GinkgoPrint` is sent directly to the ginkgo stdout AFAIU, so it is printed in both the Jenkins logs and in `test-output.log`.
This new `fmt.Errorf` will be returned as the error from the Eventually calling block. On failures, it will be printed in the Jenkins logs and in `test-output.log` as well. There's one small difference between the two :point_down: 

> Also, won't they be printed only after the last Eventually retry?

Yes. My impression is that this is what matters. We want to know what the reason was when it eventually failed. The intermediate reasons just tell us the state might not have been ready yet; that's expected for an Eventually.